test/cpp/security/tls_certificate_verifier_test.cc

```diff
@@ -161,6 +161,32 @@ TEST(TlsCertificateVerifierTest,
   EXPECT_EQ(sync_status.error_message(), "Hostname Verification Check failed.");
 }
 
+TEST(TlsCertificateVerifierTest, NoOpCertificateVerifierSucceedsWithVerifiedRootCertSubject) {
+  grpc_tls_custom_verification_check_request request;
+  memset(&request, 0, sizeof(request));
+  char* expected_subject = const_cast<char*>("CN=testca,O=Internet Widgits Pty Ltd,ST=Some-State,C=AU");
+  request.peer_info.verified_root_cert_subject = expected_subject;
+  auto verifier = std::make_shared<NoOpCertificateVerifier>();
+  TlsCustomVerificationCheckRequest cpp_request(&request);
+  EXPECT_EQ(cpp_request.verified_root_cert_subject(), expected_subject);
+  grpc::Status sync_status;
```

Comment:
Why is it necessary to call verifier->Verify here and below?

Reply:
It's not necessary, I was just using the structure from other tests.
The real thing this is testing is whether or not the `TlsCustomVerificationCheckRequest` is properly populated from the `grpc_tls_custom_verification_check_request`.

The reason I didn't take it out is that calling Verify with an unset root cert does confirm that it doesn't cause a seg fault or something like that, so I figured it might have some value to leaving it in

Reply:
It may be worth considering adding a test with an external verifier and make sure the root subject can be accessed inside the Verify() call

Reply:
Good call, makes sense.